x-pack/plugins/security_solution/public/detections/components/rules/related_integrations/utils.tsx

```diff
@@ -0,0 +1,113 @@
+          integrationName ? `?integration=${integrationName}` : ''
+        }`
+      : `${basePath}/app/integrations/detail/${packageName}`;
+  return (
+    <EuiLink href={integrationURL} target="_blank" data-test-subj={'integrationLink'}>
+      {integrationTitle}
+    </EuiLink>
+  );
+};
+
+export interface IntegrationDetails extends InstalledIntegration {
+  target_version: string;
+  version_satisfied: boolean;
+  is_installed: boolean;
+}
+
+/**
+ * Given an array of integrations and an array of installed integrations this will return an
+ * array of integrations augmented with install details like targetVersion, and `version_satisfied`
+ * has
+ * @param integrations
+ * @param installedIntegrations
+ */
+export const getInstalledRelatedIntegrations = (
+  integrations: RelatedIntegrationArray,
+  installedIntegrations: InstalledIntegrationArray | undefined
+): IntegrationDetails[] => {
+  const integrationDetails: IntegrationDetails[] = [];
+
+  integrations.forEach((i: RelatedIntegration) => {
+    const match = installedIntegrations?.find(
+      (installed) =>
+        installed.package_name === i.package && installed?.integration_name === i?.integration
+    );
+
+    if (match != null) {
+      // Version check e.g. fleet match `1.2.3` satisfies rule dependency `~1.2.1`
+      const versionSatisfied = semver.satisfies(match.package_version, i.version);
+      const packageVersion = versionSatisfied
+        ? i.version
+        : semver.valid(semver.coerce(i.version)) ?? '';
+      integrationDetails.push({
+        ...match,
+        target_version: packageVersion,
+        version_satisfied: versionSatisfied,
+        is_installed: true,
+      });
+    } else {
+      const packageVersion = semver.valid(semver.coerce(i.version)) ?? '';
+      // TODO: Add `title` to RelatedIntegration (or fetch from Fleet API) so we can accurately display the integration pretty name
+      const integrationTitle =
+        i.integration != null ? `${capitalize(i.package)} ${capitalize(i.integration)}` : undefined;
+      integrationDetails.push({
+        package_name: i.package,
+        package_title: capitalize(i.package),
+        package_version: packageVersion,
+        integration_name: i.integration,
+        integration_title: integrationTitle,
+        target_version: packageVersion,
+        version_satisfied: false,
+        is_enabled: false,
+        is_installed: false,
+      });
+    }
+  });
+
+  return integrationDetails.sort((a, b) => {
+    if (a.integration_title != null && b.integration_title != null) {
+      return a.integration_title.localeCompare(b.integration_title);
+    } else if (a.integration_title != null) {
+      return a.integration_title.localeCompare(b.package_title);
+    } else if (b.integration_title != null) {
+      return a.package_title.localeCompare(b.integration_title);
+    } else {
+      return a.package_title.localeCompare(b.package_title);
+    }
+  });
```

Comment:
Nit: Could this do the same?

```ts
  return integrationDetails.sort((a, b) => {
    const aTitle = a.integration_title ?? a.package_title;
    const bTitle = b.integration_title ?? b.package_title;
    return aTitle.localeCompare(bTitle);
  });
```

Also, if we always set `integration_title` in `IntegrationDetails`, could be even simpler:

```ts
  return integrationDetails.sort((a, b) => {
    return a.integration_title.localeCompare(b.integration_title);
  });
```

Then, in `getIntegrationLink` we wouldn't need to do ?? in `const integrationTitle = integration.integration_title ?? integration.package_title;`

Reply:
Yeah I think we can always set `integrationTitle` to clean this up and then keep `integrationName` `undefined` if it's a single integration package (i.e. `system` package).